src/index.js

```diff
@@ -1,16 +1,170 @@
+import { makeStyles } from "@material-ui/core/styles";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
 
 import "./styles.css";
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Hello CodeSandbox</h1>
-      <h2>Start editing to see some magic happen!</h2>
-    </div>
-  );
+let TABLEROWS = [];
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: "100%",
+    marginTop: theme.spacing(3),
+    overflowX: "auto"
+  },
+  table: {
+    minWidth: 650
+  }
+}));
+
+function createData(name, value) {
+  return { name, value };
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    // this.state = { resultObj: {}, show: false };
+    this.state = { resultObj: { message: "", error: false } };
+    this.inputRef = React.createRef();
+    this.processInput = this.processInput.bind(this);
+    // this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  processInput() {
+    this.setState({
+      resultObj: this.validateCourseString(this.inputRef.current.value)
+    });
+  }
+
+  validateCourseString(str) {
+    let departmentName = "";
+    let courseNumber = "";
+    let semesterName = "";
+    let year = "";
+    let i = 0;
+
+    let resultFromCourseValidation = CourseUtils.validateCourse(
+      str,
+      i,
+      departmentName,
+      courseNumber
+    );
+
+    // console.log(resultFromCourseValidation);
+    if (resultFromCourseValidation.error === true) {
+      return resultFromCourseValidation;
+    } else {
+      str = resultFromCourseValidation.str;
+      courseNumber = resultFromCourseValidation.courseNumber;
+      departmentName = resultFromCourseValidation.departmentName;
+      i = resultFromCourseValidation.i;
+    }
+    // Check if space between Course and Semester Strings occur
+    if (i >= str.length) {
+      return {
+        message:
+          "Semester Information missing after Department+Course Information",
+        error: true
+      };
+    }
+    if (str[i++] !== " ") {
+      return {
+        message:
+          "Space required between Department+Course info and Semester+Year info",
+        error: true
+      };
+    }
+    // check if invalid semester+year info i.e. either letter or digit
+    if (CommonUtils.isLetter(str[i]) === false && isNaN(str[i]) === true) {
+      return {
+        message: "Invalid character in Semester+Year info with: " + str[i],
+        error: true
+      };
+    }
+
+    // console.log(SemesterUtils);
+
+    let resultFromSemesterValidation = SemesterUtils.validateSemesterData(
+      str,
+      semesterName,
+      year,
+      i
+    );
+
+    if (resultFromSemesterValidation.error === true) {
+      return resultFromSemesterValidation;
+    } else {
+      str = resultFromSemesterValidation.str;
```

Comment:
how is this str different from the str originally given to the function? Can we not accomplish the same with only passing the current index of the str?